fakes/image.go

```diff
@@ -35,6 +35,27 @@ func NewImage(name, topLayerSha string, identifier imgutil.Identifier) *Image {
 		architecture:  "amd64",
 	}
 }
+func NewWindowsImage(name, topLayerSha string, identifier imgutil.Identifier) *Image {
+	return &Image{
+		labels:        map[string]string{},
+		env:           map[string]string{},
+		topLayerSha:   topLayerSha,
+		identifier:    identifier,
+		name:          name,
+		cmd:           []string{"initialCMD"},
+		layersMap:     map[string]string{},
+		prevLayersMap: map[string]string{},
+		createdAt:     time.Now(),
+		savedNames:    map[string]bool{},
+		os:            "windows",
+		osVersion:     "",
+		architecture:  "amd64",
+	}
+}
+
+func NewLinuxImage(name, topLayerSha string, identifier imgutil.Identifier) *Image {
+	return NewImage(name, topLayerSha, identifier)
```

Comment:
I guess we're planning to delete `NewImage` at some point. Did we create an issue for that? Maybe it's worth mentioning it in this issue.

Reply:
seems like we might want an unexported `newImage` that sets everything besides `os`. `NewWindowsImage` and `NewLinuxImage` can call that helper and then set `os`. I worry about drift/bugs if the shared setup is duplicated in `NewLinuxImage` and `NewWindowsImage`. 

Another option: we could refactor NewImage to take option functions that would include a `WithOS` option. That might be best. We currently have many tests that pass `""` and `nil` params into `fakes.NewImage`. Options functions would allow us to set only the relevant fields for each test and default everything else.

Reply:
@ekcasey, your ideas are great. I thought that we need to keep `NewImage` since some old interface is use it as is.
If I'm wrong, I totally agree with you! I think that I prefer your first option but I'm also fine with the second one.

Reply:
> I thought that we need to keep NewImage since some old interface is use it as is

If we get rid of `NewImage` or change its interface, we will have to update all the tests when we bump to the latest `imgutil` in either `lifecycle ` or `pack`. However, we should probably bite the bullet and do it if the new interface is the right choice in the long run.

Reply:
Sounds good! I think it's worth doing it!

Reply:
Ya that sound much better than what I have here. 👍

Reply:
> This ^ is probably another good reason to migrate all configuration to option functions or setters (and away from positional params). That way it is easy to extend or change in the future without breaking compilation for every test.

🖐️ for option functions over setters. Mutability is always a good source of bugs.